drivers/console/ipm_console_receiver.c

```diff
@@ -66,51 +69,29 @@ static void ipm_console_thread(void *arg1, void *arg2, void *arg3)
 			++pos;
 		}
 
-		/* ISR may have disabled the channel due to full buffer at
-		 * some point. If that happened and there is now room,
-		 * re-enable it.
-		 *
-		 * Lock interrupts to avoid pathological scenario where
-		 * the buffer fills up in between enabling the channel and
-		 * clearing the channel_disabled flag.
-		 */
-		if (driver_data->channel_disabled &&
-		    ring_buf_space_get(&driver_data->rb)) {
-			key = irq_lock();
-			ipm_set_enabled(driver_data->ipm_device, 1);
-			driver_data->channel_disabled = 0;
-			irq_unlock(key);
-		}
 	}
 }
 
 static void ipm_console_receive_callback(void *context, u32_t id,
 					 volatile void *data)
 {
 	struct device *d;
-	struct ipm_console_receiver_runtime_data *driver_data;
+	struct ipm_console_receiver_runtime_data *volatile driver_data;
```

Comment:
The volatile is incorrect here.  This delcares a volatile **pointer** (i.e. it tells the compiler that the memory where the local driver_data variable is stored may change and must not be optimized into a register, which doesn't make sense because no other thread is going to be mutating our stack like that).  You want a pointer to a volatile struct instead:

    volatile struct ipm_console_receiver_runtime_data *driver_data;

Reply:
Yes, you are right, I didn't get the right usage of volatile in that place, and above change should be the cause that make the CI test fail.